internal/catalog/internal/controllers/endpoints/controller_test.go

```diff
@@ -185,6 +185,51 @@ func TestWorkloadToEndpoint_AllAddressesFiltered(t *testing.T) {
 	require.Nil(t, workloadToEndpoint(service, data))
 }
 
+func TestWorkloadToEndpoint_MissingWorkloadProtocol(t *testing.T) {
+	// This test checks that when a workload is missing its protocol,
+	// we will default to service's protocol.
+
+	service := &pbcatalog.Service{
+		Ports: []*pbcatalog.ServicePort{
+			{TargetPort: "http", Protocol: pbcatalog.Protocol_PROTOCOL_HTTP},
```

Comment:
~ Would it make sense to call the port here and in the workload something other than `http`, so we can be sure the test is only relying on the `Protocol` value? (thinking about potential other defaulting magic in the future)

Reply:
Done!